### PR TITLE
Add filter to avoid processing more than one POST requests in a release

### PR DESCRIPTION
**Pull request by manekenpix**
Co-authored-by: Calvin <calvinho125@gmail.com>

## Type of Change

- [x] **Bugfix**: Change which fixes an issue
- [ ] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

GitHub's webhooks send 3 POST requests whenever there's a new release. After some testing, I can confirm that our autodeployment server on `prod` crashes trying to process the 3 POST requests in parallel.
This PR filters out the requests that don't need to be considered.

Info about release's actions [here](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#release).

## Checklist

- [x] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

### CODE REVIEW

File: tools/autodeployment/server.js [hunk 1]
```diff
@@ -83,6 +83,24 @@ if (!(DEPLOY_TYPE === 'staging' || DEPLOY_TYPE === 'production')) {
...
 }
 
+/**
+ * GitHub's webhooks send 3 POST requests with different actions whenever a release event takes place: created, published and released.
+ * To prevent the autodeployment server from processing the 3 requests (and eventually crashing),
+ * we need to only allow the request whose action is 'published'.
+ * More information about release events here:
+ * https://docs.github.com/en/actions/reference/events-that-trigger-workflows#release
+ * The filter combines checking for staging or production build types and also for the right type of action for a release event in production builds
+ * @param {String} name - Repository name (should be 'telescope')
```
Comment (humphd) [inline tools/autodeployment/server.js:93]: string vs String for type info. Related, we should think about using doc typing more in backend.

**Review by humphd — APPROVED** [review]
One nit, great work tracking this down.

File: tools/autodeployment/server.js (hunk 1 above)
Comment (manekenpix) [reply tools/autodeployment/server.js:93]: Should I also change the types in lines 88 and 89 [here](https://github.com/Seneca-CDOT/telescope/blob/f35311a09e179223ea92926493c91aac9897c7d3/tools/autodeployment/server.js#L88)?

Comment (humphd) [reply tools/autodeployment/server.js:93]: Yeah, do it throughout.  Built-in types should be lowercase.

**Review by c3ho — DISMISSED** [review]

**Comment by chrispinkney** [comment]
Neat!

**Review by c3ho — APPROVED** [review]

**Review by PedroFonsecaDEV — APPROVED** [review]

**Pull request merged**